### DISCUSSION TITLE
Convert coin address from CoinType

### DISCUSSION THREAD

**Discussion by Drizzle210** [aptos-labs/aptos-developer-discussions#292] [Questions]
### Describe your question in detail.

![image](https://github.com/aptos-labs/aptos-developer-discussions/assets/55441313/d17580e9-c075-4b90-b323-486ac8ee51ab)
I want to get balance but can't get CoinType from coin address ? How can i do it ?

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by gregnazario** [comment]
## Retrieving balances of a coin
Balances of a coin can be tracked specifically by the user who's balance it is, rather than an address of the token:

So, if my account was `0x12345`, and I wanted the balance of `0x1::aptos_coin::AptosCoin` in that account I would run:
```move
  coin::balance<0x1::aptos_coin::AptosCoin>(0x12345)
```

**Comment by 0xbe1** [comment]
Not possible. a CoinType has to be passed in.
